.github/workflows/create-release.yml

```diff
@@ -6,10 +6,29 @@ on:
       milestone_title:
         description: 'Milestone title'
         required: true
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
```

Comment:
I'm not sure I understand [the docs](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-activity-types-and-filters-with-multiple-events) completely, but it looks like this will trigger the "Create release" workflow on every merge into master and on every PR, **in addition** to every day at 11am? Is there a reason for wanting both?

Reply:
To clarify, I was suggesting that maybe the daily run itself is sufficient (but I haven't done releases before, so @joshuacwnewton's opinion counts for a lot more!). I agree that regularly checking dependencies is good!

Reply:
In 99% of cases, the "merge into master" run is probably overkill, yes! 

* Our PRs already have a requirement for being up to date with master, so we already effectively test the build on master right before the PR is merged, anyway.
* If somehow there ends up being some quirk or issue that would only reveal itself on `master` branch specifically, then we would see at most a ~24h delay before finding out that something had broke (because of the cron job).

At the same time, maybe that one extra run is still useful feedback? Say we merge a PR at noon, then notice the build fails at 1pm, and then fix it by 2pm... it's probably an unlikely scenario, but it would save us a day's worth of delay. :thinking:

Reply:
In fact, this PR is a good example of a case where keeping the `master` check is useful?

Since I had to do a little hacky branch workaround to test the changes in this PR (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3678#discussion_r799585067), I'm still not 100% sure if this workflow will work on `master`. I'm 99% sure they will? But just in case there's something I've missed, it will be nice to get immediate feedback.